shell/common/gin_converters/blink_converter.cc

```diff
@@ -56,43 +57,73 @@ struct Converter<char16_t> {
   }
 };
 
-template <>
-struct Converter<blink::WebInputEvent::Type> {
-  static bool FromV8(v8::Isolate* isolate,
-                     v8::Handle<v8::Value> val,
-                     blink::WebInputEvent::Type* out) {
-    std::string type = base::ToLowerASCII(gin::V8ToString(isolate, val));
-    if (type == "mousedown")
-      *out = blink::WebInputEvent::Type::kMouseDown;
-    else if (type == "mouseup")
-      *out = blink::WebInputEvent::Type::kMouseUp;
-    else if (type == "mousemove")
-      *out = blink::WebInputEvent::Type::kMouseMove;
-    else if (type == "mouseenter")
-      *out = blink::WebInputEvent::Type::kMouseEnter;
-    else if (type == "mouseleave")
-      *out = blink::WebInputEvent::Type::kMouseLeave;
-    else if (type == "contextmenu")
-      *out = blink::WebInputEvent::Type::kContextMenu;
-    else if (type == "mousewheel")
-      *out = blink::WebInputEvent::Type::kMouseWheel;
-    else if (type == "keydown")
```

Comment:
I know there was some discussion about this already, so my thoughts may be redundant, but adding them just in case 🙂 

I think this is why passing a `keyDown` event previously triggered `before-input-event` the code that fires it [explicitly looks for `kRawKeyDown`](https://github.com/electron/electron/blob/3a3be23f0e2a6d106f9f480d4c335383f4fb4632/shell/browser/api/electron_api_web_contents.cc#L1256), so this translation from `keydown` to `rawKeyDown` was making the necessary connection into that event handler.

If that behavior is changed so that `keyDown` is *not* translated to `rawKeyDown`, that seems worthy of a note in breaking changes, and maybe a warning of some sort if someone manually fires a `keyDown` input event?

Reply:
Yeah... this is a bit of a thorny one. It's disappointing that the old API silently translated `keyDown` to `rawKeyDown`. The underlying event that `before-input-event` triggers off only triggers on `rawKeyDown` and `keyUp` (it suppresses the `char` event that comes after `rawKeyDown`: https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/render_widget_host_impl.cc;l=1778-1779;drc=27dae45c1ad0b634bbec0292c66b9873bf859fd0).

It seems like the difference between `rawKeyDown` and `keyDown` is explained [here](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/public/mojom/input/input_event.mojom;l=54-59;drc=5abffc9bbab0296c39d03cc4b894388a0b7f3258)—basically, `keyDown` === `rawKeyDown` + `char`. It says there that it varies by platform and gives Android as an example of a platform that uses `keyDown`. I'm not sure if any of the platforms we support use `keyDown` vs `rawKeyDown` + `char`, but it is definitely possible to send the `keyDown` event via DevTools protocol, or, if we choose to permit it, via `sendInputEvent`.

One option here could be to keep this translation in place for `sendInputEvent` / `before-input-event`, making it so that `rawKeyDown` appears as `keyDown` in `before-input-event` and `keyDown` gets translated to `rawKeyDown` from `sendInputEvent`. The downside there is that it would be impossible to use `sendInputEvent` to generate a `keyDown` event, and likewise impossible to detect a `keyDown` event in `before-input-event`. HOWEVER, these things were already impossible! So maybe that's fine?

It's a bit annoying that this would be inconsistent with the new `input-event` API... but maybe worth it for backwards compatibility.